services/motion/builtin/builtin.go

```diff
@@ -237,71 +247,170 @@ func (ms *builtIn) MoveOnGlobe(
 	}
-	kinematicsOptions.GoalRadiusMM = math.Min(motionCfg.PlanDeviationM*1000, 3000)
-	kinematicsOptions.HeadingThresholdDegrees = 8
 
-	plan, kb, err := ms.planMoveOnGlobe(
-		ctx,
-		componentName,
-		destination,
-		movementSensorName,
-		obstacles,
-		kinematicsOptions,
-		extra,
-	)
+	planRequest, kb, err := ms.newMoveOnGlobeRequest(ctx, componentName, destination, movementSensor, obstacles, motionCfg, extra)
 	if err != nil {
-		return false, fmt.Errorf("error making plan for MoveOnMap: %w", err)
+		return false, err
 	}
-	// execute the plan
-	for i := 1; i < len(plan); i++ {
-		ms.logger.Info(plan[i])
-		if inputEnabledKb, ok := kb.(inputEnabledActuator); ok {
-			if err := inputEnabledKb.GoToInputs(ctx, plan[i]); err != nil {
-				// If there is an error on GoToInputs, stop the component if possible before returning the error
-				if stopErr := kb.Stop(ctx, nil); stopErr != nil {
-					return false, errors.Wrap(err, stopErr.Error())
+
+	successChan := make(chan bool)
+	defer close(successChan)
+
+	replanChan := make(chan bool, 1)
+	defer close(replanChan)
+	replanChan <- true
+
+	errChan := make(chan error)
+	defer close(errChan)
+
+	var backgroundWorkers sync.WaitGroup
+	defer backgroundWorkers.Wait()
+
+	// helper function to manage polling functions
+	startPolling := func(ctx context.Context, period time.Duration, fn func(context.Context) error) {
+		backgroundWorkers.Add(1)
+		goutils.ManagedGo(func() {
+			ticker := time.NewTicker(period)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case <-ticker.C:
+					if err := fn(ctx); err != nil {
+						errChan <- err
+					}
+					return
 				}
-				return false, err
 			}
+		}, backgroundWorkers.Done)
+	}
+
+	var cancelCtx context.Context
+	var cancelFn context.CancelFunc
+	defer func() {
+		if cancelFn != nil {
+			cancelFn()
+		}
+	}()
+	for {
+		select {
+		case <-ctx.Done():
+			return false, ctx.Err()
+		case <-successChan:
+			return true, nil
+		case err := <-errChan:
+			return false, err
+		case <-replanChan:
+			fmt.Println("replanning")
+
+			if cancelFn != nil {
+				cancelFn()
+			}
+			backgroundWorkers.Wait()
+			cancelCtx, cancelFn = context.WithCancel(ctx)
+
+			inputs, err := kb.CurrentInputs(ctx)
+			if err != nil {
+				return false, nil
+			}
+			if len(kb.Kinematics().DoF()) == 2 {
+				inputs = inputs[:2]
+			}
+			planRequest.StartConfiguration = map[string][]referenceframe.Input{componentName.Name: inputs}
+
+			plan, err := motionplan.PlanMotion(ctx, planRequest)
+			if err != nil {
+				return false, nil
+			}
+
+			// drain the replanChan
+			for len(replanChan) > 0 {
+				<-replanChan
+			}
+			// drain the replanChan
+			for len(errChan) > 0 {
+				<-errChan
+			}
+
+			if positionPollingPeriod > 0 {
+				startPolling(cancelCtx, positionPollingPeriod, func(ctx context.Context) error {
+					// TODO: the function that actually monitors position
+					fmt.Println("position poll")
```

Comment:
[nit]: no need to print statements

Reply:
yes I will remove these now, they were useful for testing